### ISSUE TITLE
[Documentation] Convert torch model to onnx in half precision

### ISSUE THREAD

**Issue by FrancescoSaverioZuppichini** [microsoft/onnxruntime#13833]
### Describe the documentation issue

Is there an example about how to convert half precision torch models to onnx?

**Comment by tianleiwu** [comment]
Use torch.onnx.export to export models to onnx. See related documents:
https://github.com/onnx/tutorials#converting-to-onnx-format
https://pytorch.org/docs/stable/onnx.html

File an issue to pytorch (https://github.com/pytorch/pytorch) if it is not working as expected.

I suggest to try two ways: (1) directly export half model (2) load torch model as fp32 (make sure the modeling script use fp32 in computation), export it to onnx, then optimize the model (If your model is transformers, try https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/README.md), finally convert fp32 onnx model to fp16 using https://github.com/microsoft/onnxconverter-common/blob/master/onnxconverter_common/float16.py. Sometime, the second way could get better performance.

**Comment by FrancescoSaverioZuppichini** [comment]
hey @tianleiwu thanks for your reply. 

I have found a way to convert the model directly to torch, do you think I can PR a little addition to the doc? 

Pasting here, basically, as you said one can export the model in float16 directly


```python
import clip
import onnxruntime as ort
import torch


model_name = "MODEL_NAME.onnx"
model = ....

with torch.autocast("cuda", dtype=torch.float16):
    x = torch.randn(1, 3, 224, 224, device="cuda")
    # # Export the model
    torch.onnx.export(
        model,  # model being run
        x,  # model input (or a tuple for multiple inputs)
        model_name,  # where to save the model (can be a file or file-like object)
        opset_version=16,
        export_params=True,  # store the trained parameter weights inside the model file
        do_constant_folding=True,  # whether to execute constant folding for optimization
        input_names=["image"],  # the model's input names
        output_names=["output"],  # the model's output names
        dynamic_axes={
            "image": {0: "batch_size"},  # variable length axes
            "output": {0: "batch_size"},
        },
    )
    ```

**Comment by FrancescoSaverioZuppichini** [comment]
Thanks guys, btw I am trying to convert the model in half precision, but no luck so far

https://github.com/microsoft/onnxconverter-common/issues/252

**Comment by FrancescoSaverioZuppichini** [comment]
hey guys, is it possible to receive support? It looks like there is no working documentation about how to convert a model to float16 or mixed precision. Most of CV architecture have BatchNorm, thus we need a way to use mixed precision. Thanks

**Comment by natke** [comment]
Hi @FrancescoSaverioZuppichini, the converter team are looking into the issue. From the options above, are you using method (1) or (2), and can you describe what is not working?

**Comment by FrancescoSaverioZuppichini** [comment]
Hi @natke, thanks for the reply. You can see an on going issue here -> https://github.com/microsoft/onnxruntime/issues/14235

But, how do you guys do it internally? I still cannot wrap my head around it, is it something you is actually possible and was done in the past or it's still a "beta"?

**Comment by FrancescoSaverioZuppichini** [comment]
@natke I am also down for a call, feel free to DM me on [LinkedIn](https://www.linkedin.com/in/francesco-saverio-zuppichini-94659a150/) in case

**Comment by natke** [comment]
@FrancescoSaverioZuppichini Since both of the issues you have identified come from the onnxconverter_common libs, that is the best place to continue the discussion. I'll close this issue for now. Please re-open it, or open a new one based on what you discuss with the converters team

**Comment by FrancescoSaverioZuppichini** [comment]
Thanks a lot for the support @natke
